tests/installation/disk_activation.pm

```diff
@@ -26,79 +33,91 @@ sub run {
-        assert_screen 'zfcp-activated';
-        send_key $cmd{next};
-        wait_still_screen 5;
+            assert_screen 'zfcp-disk-management';
+            assert_screen 'zfcp-activated';
+            send_key $cmd{next};
+            wait_still_screen 5;
+        }
     }
     else {    # use default DASD as install disk
         assert_screen 'disk-activation', 15;
         send_key 'alt-d';    # configure DASD disk
-        assert_screen 'dasd-disk-management';
-
-        # we need to type backspace to delete the content of the input field in textmode
-        if (check_var("VIDEOMODE", "text")) {
-            send_key 'alt-m';    # minimum channel ID
-            for (1 .. 9) { send_key "backspace"; }
-            type_string '0.0.0150';
-            send_key 'alt-x';    # maximum channel ID
-            for (1 .. 9) { send_key "backspace"; }
-            type_string '0.0.0150';
-            send_key 'alt-f';    # filter button
-            assert_screen 'dasd-unselected';
-            send_key 'alt-s';    # select all
-            assert_screen 'dasd-selected';
-            send_key 'alt-a';    # perform action button
-            assert_screen 'action-list';
-            send_key 'alt-a';    # activate
+                             # workaround for bsc#1055871
+        if (!check_screen('dasd-disk-management', 5)) {
+            workaround_broken_disk_activation;
         }
         else {
-            send_key 'alt-m';    # minimum channel ID
-            type_string '0.0.0150';
-            send_key 'alt-x';    # maximum channel ID
-            type_string '0.0.0150';
-            send_key 'alt-f';    # filter button
-            assert_screen 'dasd-unselected';
-            send_key 'alt-s';    # select all
-            assert_screen 'dasd-selected';
-            send_key 'alt-a';    # perform action button
-            assert_screen 'action-list';
-            send_key 'a';        # activate
-        }
+            assert_screen 'dasd-disk-management';
 
-        # sometimes it happens, that the DASD is in a unstable state, so
-        # if the systems wants to format the DASD by itself, do it.
-        if (check_screen 'dasd-format-device', 10) {    # format device pop-up
-            send_key 'alt-o';                           # continue
-            format_dasd;
-        }
-
-        # format DASD if the variable is that, because we format it usually pre-installation
-        elsif (get_var('FORMAT_DASD_YAST')) {
-            send_key 'alt-s';                           # select all
-            assert_screen 'dasd-selected';
-            send_key 'alt-a';                           # perform action button
-            if (check_screen 'dasd-device-formatted') {
+            # we need to type backspace to delete the content of the input field in textmode
+            if (check_var("VIDEOMODE", "text")) {
+                send_key 'alt-m';    # minimum channel ID
+                for (1 .. 9) { send_key "backspace"; }
```

Comment:
Would be nice, to use send_key_until_needle_matches here, instead of a loop.

Reply:
true, but not part of this pull request
This is old code which worked for quite a while now, just appears as new because of the new indentation